store/sqlstore/scheme_supplier.go

```diff
@@ -161,6 +165,10 @@ func (s *SqlSchemeStore) createScheme(scheme *model.Scheme, transaction *gorp.Tr
 			SchemeManaged: true,
 		}
 
+		if scheme.Scope == model.SCHEME_SCOPE_CHANNEL {
+			channelAdminRole.Permissions = []string{}
```

Comment:
Can switch to `filterModerated` here if the logic will keep the permissions list empty for the channel admin.

Reply:
The only way this is feasible is by telling this function what type (ie admin, user, or guest) role the function is dealing with. It couldn't be done as a role method either because role do not know their type.

Reply:
Add a comment for this special case.